### PR TITLE
drivers/clock_control: stm32wl: Changes around use of  HSE as input clock

### LINKED ISSUE #37260: STM32WL does not support HSE as RCC source and HSEDiv
**Describe the bug**
STM32WL does not support HSE correctly as RCC clock source. Also STM32WLxx has two div option for HSE, but there is not way to select HSEDiv. I have tested on my STM32WL55CC custom board using 32MHz crystal, and it worked with HSI, but not with HSE. For the STM32CubeWL project, I was using HSEDiv = 2.

**To Reproduce**
1. Update the board dts as following.
```
&clk_hse {
    // 32MHz external crystal
	status = "okay";
};
&pll {
	div-m = <2>;
	mul-n = <6>;
	div-p = <2>;
	div-r = <2>;
	div-q = <2>;
	clocks = <&clk_hse>;
	status = "okay";
};
```
2. build blinky sample

**Expected behavior**
It should blink but didn't at all.
Anyway, I set the `mul-n` bigger than 32, it was blinky and the blinky frequency is about 3 times and can see the corrupted log. Also there is no difference between 32 and 127 of `mul-n`
By RM0453 7.2.4 PLL, _"The PLLQCLK and PLLRCLK output frequency must not exceed 48 MHz. The PLLPCLK output frequency must not exceed 62 MHz."_, So it's not the right configuration.

**Environment**
 - OS: Ubuntu 18.04 LTS
 - Toolchain Zephyr SDK 0.12.4
 - ZephyrRTOS v2.6.99

### PR DESCRIPTION
**Pull request by erwango**
This change around use of HSE on STM32WL is two fold:
- Make use of https://github.com/zephyrproject-rtos/hal_stm32/pull/111 which fixes a bug in STM32Cube LL API. (This is a temporary patch as official bug has been filled internally. Reference provided)
- Add a dedicated binding for STM32WL HSE clock and implement related changes

Fixes #37260

### CODE REVIEW

**Comment by erwango** [comment]
Getting false positives on the compliance check:
```
-:27: WARNING:IS_ENABLED_CONFIG: IS_ENABLED(STM32_HSE_TCXO) is normally used as IS_ENABLED(CONFIG_STM32_HSE_TCXO)
#27: FILE: drivers/clock_control/clock_stm32_ll_common.c:499:
+	if (IS_ENABLED(STM32_HSE_TCXO)) {

-:30: WARNING:IS_ENABLED_CONFIG: IS_ENABLED(STM32_HSE_DIV2) is normally used as IS_ENABLED(CONFIG_STM32_HSE_DIV2)
#30: FILE: drivers/clock_control/clock_stm32_ll_common.c:502:
+	if (IS_ENABLED(STM32_HSE_DIV2)) {

-:48: WARNING:IS_ENABLED_CONFIG: IS_ENABLED(STM32_HSE_DIV2) is normally used as IS_ENABLED(CONFIG_STM32_HSE_DIV2)
#48: FILE: drivers/clock_control/clock_stm32_ll_common.c:530:
+	if (IS_ENABLED(STM32_HSE_DIV2)) {

-:65: WARNING:IS_ENABLED_CONFIG: IS_ENABLED(STM32_HSE_TCXO) is normally used as IS_ENABLED(CONFIG_STM32_HSE_TCXO)
#65: FILE: drivers/clock_control/clock_stm32_ll_common.c:558:
+		if (IS_ENABLED(STM32_HSE_TCXO)) {

-:72: WARNING:IS_ENABLED_CONFIG: IS_ENABLED(STM32_HSE_BYPASS) is normally used as IS_ENABLED(CONFIG_STM32_HSE_BYPASS)
#72: FILE: drivers/clock_control/clock_stm32_ll_common.c:563:
+		if (IS_ENABLED(STM32_HSE_BYPASS)) {
```

File: drivers/clock_control/clock_stm32_ll_common.c [hunk 1]
```diff
@@ -542,18 +554,18 @@ int stm32_clock_control_init(const struct device *dev)
...
 	/* Enable HSE if not enabled */
 	if (LL_RCC_HSE_IsReady() != 1) {
-		/* Check if need to enable HSE bypass feature or not */
-		if (IS_ENABLED(STM32_HSE_BYPASS)) {
 #ifdef CONFIG_SOC_SERIES_STM32WLX
+		if (IS_ENABLED(STM32_HSE_TCXO)) {
 			LL_RCC_HSE_EnableTcxo();
-		} else {
-			LL_RCC_HSE_DisableTcxo();
+		}
```
Comment (ABOSTM) [inline drivers/clock_control/clock_stm32_ll_common.c:560]: `else { LL_RCC_HSE_DisableTcxo(); }   ` ?  ... similar to Bypass

File: drivers/clock_control/clock_stm32_ll_common.c [hunk 2]
```diff
@@ -486,18 +487,24 @@ int stm32_clock_control_init(const struct device *dev)
 	LL_RCC_MSI_Disable();
 
 #elif STM32_PLL_SRC_HSE
-	int hse_bypass;
 
+#ifndef CONFIG_SOC_SERIES_STM32WLX
```
Comment (ABOSTM) [inline drivers/clock_control/clock_stm32_ll_common.c:491]: nit: why not removing the negation and invert #if and #else ?

**Review by ABOSTM — COMMENTED** [review]
2 minor remarks

File: drivers/clock_control/clock_stm32_ll_common.c (hunk 1 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_common.c:560]: Default configuration is TCXO disabled.
Then, the if condition is defined at build time, so there is no way we can enable then disable.

I think that the one to be fixed would be the bypass part.

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by mbolivar-nordic — APPROVED** [review]
Looks good pending merge of zephyrproject-rtos/hal_stm32#111 and update of west.yml to the final SHA

**Review by FRASTM — APPROVED** [review]

**Comment by cfriedt** [comment]
@erwango - please rebase - will merge once it's all green

**Comment by erwango** [comment]
> @erwango - please rebase - will merge once it's all green

@cfriedt compliance issue is a false positive (as mentioned her: https://github.com/zephyrproject-rtos/zephyr/pull/37286#issuecomment-888375742).
Aside this, everything is green.

**Pull request merged**